gr-fec/lib/viterbi/metrics.c

```diff
@@ -51,76 +53,81 @@ extern double erf(double x);
 /* Logarithm base 2 */
 #define gr_log2(x) (log(x)*M_LOG2E)
 
+namespace gr{
+  namespace fec{
+
 /* Generate log-likelihood metrics for 8-bit soft quantized channel
  * assuming AWGN and BPSK
  */
-void
-gen_met(int mettab[2][256],	/* Metric table, [sent sym][rx symbol] */
-	int amp,		/* Signal amplitude, units */
-	double esn0,		/* Es/N0 ratio in dB */
-	double bias,		/* Metric bias; 0 for viterbi, rate for sequential */
-	int scale)		/* Scale factor */
-{
-  double noise;
-  int s,bit;
-  double metrics[2][256];
-  double p0,p1;
-
-  /* Es/N0 as power ratio */
-  esn0 = pow(10.,esn0/10);
-
-  noise = 0.5/esn0;	/* only half the noise for BPSK */
-  noise = sqrt(noise);	/* noise/signal Voltage ratio */
-
-  /* Zero is a special value, since this sample includes all
-   * lower samples that were clipped to this value, i.e., it
-   * takes the whole lower tail of the curve
-   */
-  p1 = normal(((0-OFFSET+0.5)/amp - 1)/noise);	/* P(s|1) */
-
-  /* Prob of this value occurring for a 0-bit */	/* P(s|0) */
-  p0 = normal(((0-OFFSET+0.5)/amp + 1)/noise);
-  metrics[0][0] = gr_log2(2*p0/(p1+p0)) - bias;
-  metrics[1][0] = gr_log2(2*p1/(p1+p0)) - bias;
-
-  for(s=1;s<255;s++){
-    /* P(s|1), prob of receiving s given 1 transmitted */
-    p1 = normal(((s-OFFSET+0.5)/amp - 1)/noise) -
-      normal(((s-OFFSET-0.5)/amp - 1)/noise);
-
-    /* P(s|0), prob of receiving s given 0 transmitted */
-    p0 = normal(((s-OFFSET+0.5)/amp + 1)/noise) -
-      normal(((s-OFFSET-0.5)/amp + 1)/noise);
+    void
```

Comment:
Argh fixing whitespace in the same commit makes reading reviews hard. We should really do a mega-whitespace fix soon.

Reply:
I think most of the whitespacing "issue" stems from inserting the full block into a namespace and then I just did a autoindent to have them in the correct indentation...